core/src/main/java/org/apache/calcite/util/format/postgresql/FormatPattern.java

```diff
@@ -36,6 +59,152 @@ public interface FormatPattern {
    * @param dateTime datetime to convert
    * @return the string representation of the datetime based on the format pattern
    */
-  @Nullable String convert(ParsePosition parsePosition, String formatString,
+  public abstract @Nullable String convert(ParsePosition parsePosition, String formatString,
       ZonedDateTime dateTime);
+
+  /**
+   * Get the ChronoUnitEnum value that this format pattern represents. For example, the
+   * pattern YYYY is for YEAR.
+   *
+   * @return a ChronoUnitEnum value
+   */
+  protected abstract ChronoUnitEnum getChronoUnit();
+
+  /**
+   * Attempts to parse a single value from the input for this pattern. It will start parsing
```

Comment:
this is much better

Reply:
@mihaibudiu I see what you mean now. I have updated the return type and updated any tests that were affected.